src/renderer/lib/hooks/useAppListeners.ts

```diff
@@ -252,17 +251,32 @@ const handleDolphinExitCode = (exitCode: number | null) => {
     return null;
   }
 
-  // Dolphin returns 3 when selecting Update in game on Windows
-  if (exitCode === 3 && isWindows) {
-    return null;
-  }
-
-  if (exitCode === 0xc0000135 || exitCode === 0xc0000409) {
-    return `Required DLLs for launching Dolphin are missing. Check the Help section in the settings page to fix this issue.`;
-  }
-
-  if (exitCode === 0x7f && isLinux) {
-    return `Required libraries for launching Dolphin may be missing. Check the Help section in the settings page for guidance. Post in the Slippi Discord's linux-support channel for further assistance if needed.`;
+  switch (process.platform) {
+    case "win32": {
+      switch (exitCode) {
+        case 0x3: {
+          // returned when selecting update in game
+          return null;
+        }
+        case 0xc0000135:
+        case 0xc0000409:
+        case 0xc000007b: {
+          return "Required DLLs for launching Dolphin are missing. Check the Help section in the settings page to fix this issue.";
+        }
+        case 0xc0000005: {
+          return "Install the latest Windows update available and then restart your computer.";
+        }
+      }
+      break;
+    }
```

Comment:
Yeah nah. Move this to a separate function like `handleWindowsErrorCode(code: number): string | null` then you can have your switch statement in there. Liekwise with the linux codes.

Reply:
Also while you're at it, move those error handling functions and the `handleDolphinExitCode` function outside of the `useAppListeners` hook. We don't really need to define them inside the main hook.